src/main/java/java_echo_server/EchoServer.java

```diff
@@ -0,0 +1,28 @@
+package java_echo_server;
+
+public class EchoServer {
```

Comment:
I feel like this class is handling the echo behavior more than it is handling the server behavior, while most of the server behavior is in your `App` class. I'd consider renaming this class.